engines/scumm/gfx.cpp

```diff
@@ -1109,30 +1173,44 @@ void ScummEngine::restoreCharsetBg() {
 		_charset->_str.left = -1;
 		_charset->_left = -1;
 
-		// Restore background on the whole text area. This code is based on
-		// restoreBackground(), but was changed to only restore those parts which are
-		// currently covered by the charset mask.
+		int i;
+		int w = 8;
```

Comment:
Seems that `w` is used only within `for (i = 0; i < _gdi->_numStrips; i++) {` loop. Please move it there for reducing its scope.